lib/puppet/file_serving/content.rb

```diff
@@ -21,11 +21,9 @@ def self.from_raw(content)
     instance
   end
 
-  # BF: we used to fetch the file content here, but this is counter-productive
-  # for puppetmaster streaming of file content. So collect just returns itself
+  # This is no longer used, but is still called by the file server implementations when interacting
+  # with their model abstraction.
   def collect(source_permissions = nil)
-    return if stat.ftype == "directory"
```

Comment:
Implementation removed because it's dead code, the results aren't used anywhere. The only call to this method now comes from the inherited #find/#search calls in indirector/file_content/{file.rb, file_server.rb}.